source/retryable-reads/tests/unified/handshakeError.yml

```diff
@@ -0,0 +1,68 @@
+description: "retryable reads handshake failures"
+
+schemaVersion: "1.0"
+
+runOnRequirements:
+  - minServerVersion: "4.2"
+    topologies: [replicaset, sharded, load-balanced]
+    auth: true
+
+createEntities:
+  - client:
+      id: &client0 client0
+      useMultipleMongoses: false
+      observeEvents: [commandStartedEvent, connectionCheckOutStartedEvent]
+      uriOptions: { retryReads: true, retryWrites: true, maxPoolSize: 5 }
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &databaseName retryable-handshake-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collectionName coll
+
+initialData:
+  - collectionName: *collectionName
+    databaseName: *databaseName
+    documents:
+      - { _id: 1, x: 11 }
+      - { _id: 2, x: 22 }
+      - { _id: 3, x: 33 }
+
+tests:
+  - description: "InsertOne succeeds after retryable handshake network error"
+    operations:
+      - name: failPoint # fail the next connection establishment
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 2 }
+            data:
+              failCommands: [saslContinue, dropCollection] # use saslContinue here to avoid SDAM errors
```

Comment:
If these operations execute sequentially, what necessitates the driver checking out a second connection (which will fall during `saslContinue`) instead of re-using the first already-authed connection?

Separate question: `saslContinue` isn't used for _every_ [auth mechanism](https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst). Would this test fail if the test suite was run against a cluster using X509 authentication? If so, is that a concern that'd warrant new test format syntax in `runOnRequirement` to check the authMechanism? There may be a BC-friendly way to introduce that where `auth` can start accepting an array of permitted authMechanism strings while still allowing a boolean (i.e. `true` matches any mechanism, and `false` requires none).

Reply:
I mention the reason behind this dropCollection in another thread.

As for the `saslContinue`, this is a way to get handshakes to fail only for operations and not on monitoring connections. There's precedent for it in sdam integration tests and (where I originally got the idea from) this [lb test](https://github.com/mongodb/specifications/blob/master/source/load-balancers/tests/sdam-error-handling.yml#L187). But I do agree that being able to be specific about the authMechanism could be good for future shared coverage of those mechs. I can file a followup drivers ticket

Reply:
Feel free to resolve this after you file the separate DRIVERS ticket (if you'd like to use this thread as a reminder).

Reply:
Created: https://jira.mongodb.org/browse/DRIVERS-2043